atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTableTest.java

```diff
@@ -348,6 +357,80 @@ public void createDoesNotThrowWhenOverflowColumnAlreadyExistsAndAlterIsRan() thr
         verifyTableAltered();
     }
 
+    @Test
+    public void testIndexStrategy_conservative_sweep() {
+        for (int col = 0; col < 10; col++) {
+            TableMetadata.Builder metadataBuilder = TableMetadata.builder().sweepStrategy(SweepStrategy.CONSERVATIVE);
+            for (int colName = 0; colName < col; colName++) {
+                boolean hasOverflow = ThreadLocalRandom.current().nextBoolean();
+                metadataBuilder.singleNamedColumn(
+                        "foo" + colName, "foobar" + colName, hasOverflow ? ValueType.STRING : ValueType.FIXED_LONG);
+            }
+            TableMetadata metadata = metadataBuilder.build();
+            assertThat(OracleDdlTable.getOptimalIndexCompression(metadata)).isEqualTo(2);
+        }
+    }
+
+    @Test
+    public void testIndexStrategy_dynamic_columns() {
+        for (int col = 1; col <= 10; col++) {
+            TableMetadata.Builder metadataBuilder = TableMetadata.builder().sweepStrategy(SweepStrategy.THOROUGH);
+            List<NameComponentDescription> dynamicColumns = new ArrayList<>();
+            for (int colDesc = 1; colDesc <= col; colDesc++) {
+                boolean hasOverflow = ThreadLocalRandom.current().nextBoolean();
+                ValueType valueType =
+                        colDesc == col ? (hasOverflow ? ValueType.STRING : ValueType.FIXED_LONG) : ValueType.SHA256HASH;
```

Comment:
Might be missing something: some of these generations feel a bit more complex than required (e.g., as opposed to some helper function that has a list of types and just picks from that).